consensus/polybft/validator_set_test.go

```diff
@@ -0,0 +1,528 @@
+package polybft
+
+import (
+	"bytes"
+	"math"
+	"testing"
+
+	bls "github.com/0xPolygon/polygon-edge/consensus/polybft/signer"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestValSetIndex(t *testing.T) {
+	t.Parallel()
+
+	keys, err := bls.CreateRandomBlsKeys(5)
+	require.NoError(t, err)
+
+	addresses := []types.Address{{0x10}, {0x52}, {0x33}, {0x74}, {0x60}}
+
+	vs, err := NewValidatorSet([]*ValidatorMetadata{
+		{
+			BlsKey:      keys[0].PublicKey(),
+			Address:     addresses[0],
+			VotingPower: 10,
+		},
+		{
+			BlsKey:      keys[1].PublicKey(),
+			Address:     addresses[1],
+			VotingPower: 100,
+		},
+		{
+			BlsKey:      keys[2].PublicKey(),
+			Address:     addresses[2],
+			VotingPower: 1,
+		},
+		{
+			BlsKey:      keys[3].PublicKey(),
+			Address:     addresses[3],
+			VotingPower: 50,
+		},
+		{
+			BlsKey:      keys[4].PublicKey(),
+			Address:     addresses[4],
+			VotingPower: 30,
+		},
+	})
+	require.NoError(t, err)
+	// validate no changes to validator set positions
+	for i, v := range vs.Accounts() {
+		assert.Equal(t, addresses[i], v.Address)
+	}
+
+	proposer, err := vs.CalcProposer(0)
+	require.NoError(t, err)
+	assert.Equal(t, proposer, addresses[1])
+
+	// validate no changes to validator set positions
+	for i, v := range vs.Accounts() {
+		assert.Equal(t, addresses[i], v.Address)
+	}
+}
```

Comment:
Suggesting this change, as it would make unit test more compact (similar goes to other tests as well, feel free to rely on `testValidators` as it makes life a lot easier :slightly_smiling_face:):

```suggestion
func TestValSetIndex(t *testing.T) {
	t.Parallel()

	validators := newTestValidatorsWithAliases([]string{"A", "B", "C", "D", "E"}, []uint64{10, 100, 1, 50, 30})
	metadata := validators.getPublicIdentities()
	vs := validators.toValidatorSetWithError(t)

	// validate no changes to validator set positions
	for i, v := range vs.Accounts() {
		assert.Equal(t, metadata[i].Address, v.Address)
	}

	proposer, err := vs.CalcProposer(0)
	require.NoError(t, err)
	assert.Equal(t, proposer, metadata[1].Address)

	// validate no changes to validator set positions
	for i, v := range vs.Accounts() {
		assert.Equal(t, metadata[i].Address, v.Address)
	}
}
```

Reply:
not sure what I'm missing but this does not compile since some methods used do not exist.

Reply:
Your local branch isn't synchronized to the remote. You are lacking this commit: 55b8e2059b5f5f1a43b55e078ce7b4de484556eb.

Reply:
Your local branch isn't synchronized with the remote. You are lacking this commit: 55b8e2059b5f5f1a43b55e078ce7b4de484556eb.

Reply:
why tests were not updated with this commit https://github.com/0xPolygon/polygon-edge/commit/55b8e2059b5f5f1a43b55e078ce7b4de484556eb